pkg/cosign/image_signer.go

```diff
@@ -111,6 +134,139 @@ func (s *ImageSigner) sign(ro *options.RootOptions, refImage, secretLocation, co
+		keyRef := fmt.Sprintf("k8s://%s/%s", secret.Namespace, secret.Name)
+		keyOpts := options.KeyOpts{
+			KeyRef: keyRef,
+			PassFunc: func(bool) ([]byte, error) {
+				if password, ok := secret.Data[SecretDataCosignPassword]; ok {
+					return password, nil
+				}
+
+				return []byte(""), nil
+			},
+		}
+
+		if cosignRepository, ok := secret.Annotations[RepositoryAnnotationPrefix]; ok {
+			if err := os.Setenv(cosignRepositoryEnv, cosignRepository); err != nil {
+				return nil, fmt.Errorf("failed setting %s env variable: %w", cosignRepositoryEnv, err)
+			}
+		}
+
+		if cosignDockerMediaType, ok := secret.Annotations[DockerMediaTypesAnnotationPrefix]; ok {
+			if err := os.Setenv(cosignDockerMediaTypesEnv, cosignDockerMediaType); err != nil {
+				return nil, fmt.Errorf("failed setting %s env variable: %w", cosignDockerMediaTypesEnv, err)
+			}
+		}
+
+		registryOptions := options.RegistryOptions{KubernetesKeychain: true, Keychain: builderKeychain}
+
+		signOptions := options.SignOptions{
+			Registry:          registryOptions,
+			AnnotationOptions: options.AnnotationOptions{},
+			Upload:            true,
+			Recursive:         false,
+			TlogUpload:        false,
+		}
+
+		rootOptions := options.RootOptions{Timeout: Timeout}
+
+		if err := s.signFunc(
+			&rootOptions,
+			keyOpts,
+			signOptions,
+			[]string{imageReference}); err != nil {
+			return nil, errors.Errorf("unable to sign image with specified key from secret %v in namespace %v: %v", secret.Name, secret.Namespace, err)
+		}
+
+		reference, err := name.ParseReference(imageReference)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse reference: %w", err)
+		}
+
+		registryOpts, err := registryOptions.ClientOpts(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		signatureTag, err := s.fetchSignatureFunc(reference, registryOpts...)
+		if err != nil {
+			return nil, err
+		}
+
+		image, err := remote.Image(signatureTag, remote.WithAuthFromKeychain(builderKeychain))
+		if err != nil {
+			return nil, err
+		}
+
+		digest, err := image.Digest()
+		if err != nil {
+			return nil, err
+		}
+
+		signaturePaths = append(
+			signaturePaths,
+			v1alpha2.SignaturePath{
+				KeyName: keyRef, // TODO check if secret name is enough
+				Path:    signatureTag.Digest(digest.String()).String(),
+			},
+		)
+
+		if _, found := os.LookupEnv(cosignDockerMediaTypesEnv); found {
+			err = os.Unsetenv(cosignDockerMediaTypesEnv)
+			if err != nil {
+				return nil, fmt.Errorf("failed to cleanup environment variable %s: %s", cosignDockerMediaTypesEnv, err)
+			}
+		}
+
+		if _, found := os.LookupEnv(cosignRepositoryEnv); found {
+			err = os.Unsetenv(cosignRepositoryEnv)
+			if err != nil {
+				return nil, fmt.Errorf("failed to cleanup environment variable %s: %s", cosignRepositoryEnv, err)
+			}
+		}
+	}
+
+	return signaturePaths, nil
+}
+
+func FindCosignSecrets(ctx context.Context, client k8sclient.Interface, namespace string, serviceAccountName string) ([]corev1.Secret, error) {
```

Comment:
I think it might be a little confusing that we have 2 methods with the same name that are used in different places and do different things

Reply:
Hi there, @tomkennedy513! I've renamed it to `FetchBuilderSigningSecrets` instead.